### PR TITLE
many linter fixes

### PR DESCRIPTION
**Pull request by barabo**
### Description

The goal of this PR is to silence the pylint warnings in vscode, which will make it easier for people to make changes to the code in vscode.

There were _many_ linter issues found, which led to a few refactors which eliminated unused parameters.  The unit tests are all passing when testing manually except for 2, but both of those failing unit tests are also failing in the merge target branch - so I don't believe they were introduced by these changes.

<table>
<tr>
<th>BEFORE</th>
<th>AFTER</th>
</tr>
<tr>
<td>

image</td>

<td>

image

</td>
</tr>
</table>

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda_index/index/convert_cache.py [hunk 1]
```diff
@@ -322,22 +326,22 @@ def merge_index_cache(channel_root, output_db="merged.db"):
...
     conn = common.connect("linux-64-cache.db")
     create(conn)
-    with closing(conn):
+    with closing(conn):  # type: ignore
         convert_cache(conn, extract_cache(archive_path))
         remove_prefix(conn)
 
 
-def test():
+def _test():
```
Comment (dholth) [inline conda_index/index/convert_cache.py:344]: Is this to prevent pytest from finding the function?

File: conda_index/index/__init__.py [hunk 2]
```diff
@@ -137,14 +139,15 @@ def update_index(
...
+            log.warning(
+                "The update_index function has changed to index all subdirs at once.  "
+                "You're pointing it at a single subdir.  "
+                "Please update your code to point it at the channel root, rather than a subdir.  "
                 "Use -s=<subdir> to update a single subdir."
             )
         raise SystemExit()
 
-    channel_index = ChannelIndex(
+    channel_index = _ChannelIndex(
```
Comment (dholth) [inline conda_index/index/__init__.py:150]: We can't rename this class.

File: conda_index/index/__init__.py [hunk 3]
```diff
@@ -448,9 +452,10 @@ def _build_current_repodata(subdir, repodata, pins):
     for keep_pkg in keep_pkgs:
         if keep_pkg.fn.endswith(CONDA_PACKAGE_EXTENSION_V2):
             conda_packages[keep_pkg.fn] = repodata["packages.conda"][keep_pkg.fn]
-            # in order to prevent package churn we consider the md5 for the .tar.bz2 that matches the .conda file
-            #    This holds when .conda files contain the same files as .tar.bz2, which is an assumption we'll make
-            #    until it becomes more prevalent that people provide only .conda files and just skip .tar.bz2
+            # In order to prevent package churn, we consider the md5 for the .tar.bz2 that matches
+            # the .conda file.  This holds true when .conda files contain the same files as
+            # .tar.bz2, which is an assumption we'll make until it becomes more prevalent that
+            # people provide only .conda files and just skip .tar.bz2.
```
Comment (dholth) [inline conda_index/index/__init__.py:458]: We should check this again now that we use our _speed_ to index `.conda` and `.tar.bz2` independently.

**Comment by jaimergp** [comment]
This is a good idea, but I'd say we should also implement pre-commit hooks to enforce the linter going forward.

File: conda_index/index/__init__.py [hunk 4]
```diff
@@ -1077,7 +1076,7 @@ def json_dumps(self, data):
             return json.dumps(data, sort_keys=True, indent=2) + "\n"
 
     def _write_channeldata(self, channeldata):
-        # trim out commits, as they can take up a ton of space.  They're really only for the RSS feed.
+        # Trim out commits, as they can take up a ton of space.  They're really only for the RSS.
         for pkg, pkg_dict in channeldata.get("packages", {}).items():
             channeldata["packages"][pkg] = {
                 k: v for k, v in pkg_dict.items() if v is not None and k != "commits"
```
Comment (dholth) [inline conda_index/index/__init__.py:1082]: This seems like a legacy concern. Do any current packages specify the "commits" property? (We could search the metayaml database).

File: conda_index/index/__init__.py [hunk 5]
```diff
@@ -536,11 +542,7 @@ def index(
         Examine all changed packages under ``self.channel_root``, updating
         ``index.html`` for each subdir.
         """
-        if verbose:
-            level = logging.DEBUG
-        else:
-            level = logging.ERROR
-
+        level = verbose and logging.DEBUG or logging.ERROR
         logging_context = utils.LoggingContext(level, loggers=[__name__])
```
Comment (dholth) [inline conda_index/index/__init__.py:546]: I think this has been changed to do nothing, compared to the one we inherited from conda-build.

File: README.md [hunk 6]
```diff
@@ -1,8 +1,105 @@
...
+## Getting started
+
+### Prerequisites
+If you don't have `conda` installed yet, please start by downloading
+the latest version of [anaconda] or [miniconda3].
+
+Also remember to initialize `conda` and activate a `conda` environment.
+
+```bash
+@barabo ➜ /workspaces/conda-index (main) $ conda init bash
```
Comment (dholth) [inline README.md:18]: Should remove usernames.

File: conda_index/index/convert_cache.py (hunk 1 above)
Comment (barabo) [reply conda_index/index/convert_cache.py:348]: I don't remember if that was a factor, but I tend to use leading underscores for function that I don't want referenced from an external location.

**Pull request closed without merging**